control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -471,3 +499,99 @@ func ReconcileServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, ownerRef c
 
 	return nil
 }
+
+func cvoVolumeKubeAPIAccess() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "kube-api-access",
+	}
+}
+
+func buildCVOVolumeKubeAPIAccess(v *corev1.Volume) {
```

Comment:
why would all this be needed vs just setting the SA in the deployment spec?

Reply:
This was mainly done to keep it similar to the standalone OpenShift deployment. There is an explanation for why it's needed in the standalone OpenShift. However, let me go investigate it more again, and I'll try to reduce it here in HyperShift because of the different constraints.